posts/zzz_DO_NOT_EDIT_introducing.../introducing_sdtm.oak.qmd

```diff
@@ -0,0 +1,254 @@
+    "hardcode_ct",
+    "hardcode_no_ct",
+    "condition_add"
+  ),
+  `Description` = c(
+    paste(
+      "One-to-one mapping between the raw source and a target",
+      "SDTM variable that has no controlled terminology restrictions.",
+      "Just a simple assignment",
+      "statement."
+    ),
+    paste(
+      "One-to-one mapping between the raw source and a target ",
+      "SDTM variable that is subject to controlled terminology restrictions.",
+      "A simple assign statement and applying controlled terminology.",
+      "This will be used only if the SDTM variable has an associated",
+      "controlled terminology."
+    ),
+    paste(
+      "One-to-one mapping between the raw source and a target that involves ",
+      "mapping a Date or time or datetime component. This mapping algorithm",
+      "also takes care of handling unknown dates and converting them into.",
+      "ISO8601 format."
+    ),
+    paste(
+      "Mapping a hardcoded value to a target SDTM variable that is subject to terminology restrictions.",
+      "This will be used only if the SDTM variable has an associated",
+      "controlled terminology."
+    ),
+    paste(
+      "Mapping a hardcoded value to a target SDTM variable that has no terminology restrictions."
+    ),
+    paste(
+      "Algorithm that is used to filter the source data and/or target domain",
+      "based on a condition. The mapping will be applied only if the condition is met.",
+      "The filter can be applied either at the source dataset or at target dataset or both.",
+      " This algorithm has to be used in conjunction with other algorithms, that is if the",
+      " condition is met perform the mapping using algorithms like assign_ct,",
+      "assign_no_ct, hardcode_ct, hardcode_no_ct, assign_datetime."
+    )
+  ),
+  `Example` = c(
+    paste(
+      "MH.MHTERM<br>",
+      "AE.AETERM"
+    ),
+    paste("VS.VSPOS<br>", "VS.VSLAT"),
+    paste("MH.MHSTDTC<br>", "AE.AEENDTC"),
+    paste(
+      "MH.MHPRESP = 'Y'<br>",
+      "<br>VS.VSTEST = 'Systolic Blood Pressure'<br>",
+      "<br>VS.VSORRESU = 'mmHg'<br>"
+    ),
+    paste(
+      "FA.FASCAT = 'COVID-19 PROBABLE CASE'<br>",
+      "<br>CM.CMTRT = 'FLUIDS'"
+    ),
+    paste(
+      "If If MDPRIOR == 1 then CM.CMSTRTPT = 'BEFORE'.<br>",
+      "<br>VS.VSMETHOD when VSTESTCD = 'TEMP'<br>",
+      "<br>If collected value in raw variable DOS is numeric then CM.CMDOSE<br>",
+      "<br>If collected value in raw variable MOD is different to CMTRT then map to CM.CMMODIFY"
+    )
+  ), stringsAsFactors = TRUE
+)
+knitr::kable(algorithms)
+```
+
+Here is an example of reusing an algorithm across multiple domains,
+variables, and also to a non-standard
+
+![](reusable_algorithms.jpg){width="600"}
+
+# Functions and Parameters
+
+All the aforementioned algorithms are implemented as R functions, each accepting the raw dataset, raw variable, target SDTM dataset, and target SDTM variable as parameters.
+
+```{r}
```

Comment:
```suggestion
```{r, message = FALSE}
```

(as otherwise you get the loading messages:)

![image](https://github.com/user-attachments/assets/52f27b2b-98fb-4cf4-831c-70de293348e2)